src/ClassDiscovery.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Http\Discovery;
+
+/**
+ * Registry that based find results on class existence
+ *
+ * @author David de Boer <david@ddeboer.nl>
+ */
+abstract class ClassDiscovery implements Discovery
```

Comment:
Fine, but in that case the parameters should always be the same. See comment bellow.